bgcval2/functions/circulation.py

```diff
@@ -0,0 +1,389 @@
+
+    drake = np.sum(velo*e3u*e2u_drake*umask_drake)*1.e-6
+    
+    return drake
+
+
+
+drakedetails     = {}
+def loadDrakeDetails(gridfn):
+    global drakedetails
+    print("cmip5DrakePassage:\topening grid file", gridfn)    
+    nc = dataset(gridfn,'r')
+    if 'Drake_A' in list(nc.variables.keys()): 
+        drakeKey = 'Drake_A'
+    elif 'Drake' in list(nc.variables.keys()): 
+        drakeKey = 'Drake'
+            
+    Drake_A = nc.variables[drakeKey][:]        
+    tmask     = nc.variables['tmask'][:]
+    drakedetails[(gridfn,'Drake_A')] = Drake_A
+    drakedetails[(gridfn,'tmask')]     = tmask
+    nc.close()
+       
+ 
+def cmip5DrakePassage(nc,keys,**kwargs):
+    if 'gridfile' not in list(kwargs.keys()):
+        raise AssertionError("drakePassage:\t Needs an `gridFile` kwarg to run calculation.")    
+    gridFile_u =     kwargs['gridfile']
+    try:    
+        drakexsectArea     = drakedetails[(gridFile_u,'Drake_A')]
+        tmasku        = drakedetails[(gridFile_u,'tmask')]
+    except:
+        loadDrakeDetails(gridFile_u)
+        drakexsectArea     = drakedetails[(gridFile_u,'Drake_A')]
+        tmasku        = drakedetails[(gridFile_u,'tmask')]
+
+    drakexsectArea = np.ma.masked_where((drakexsectArea==0.) + (tmasku==1),drakexsectArea)
+    velo = nc.variables[keys[0]]#[:]
+    out = []
+    if velo.ndim==4:
+        for t in np.arange(velo.shape[0]):
+            #print 'cmip5DrakePassage',t, velo[t].shape, drakexsectArea.shape, tmasku.shape, gridFile_u
+            drk = velo[t] *drakexsectArea 
+            
+            drk = np.ma.masked_where((drk==0.) + (tmasku==1.) +drk.mask, drk)
+            xsection = drk.sum(2)*1.e-6
+            totaldrk = drk.sum()*1.e-6                        
+            plots=0
+            if plots:                        
+                from matplotlib import pyplot
+                print(drk.shape,xsection.shape, velo.shape, xsection.sum(),totaldrk)                        
+                ax1 = pyplot.subplot(411)
+                pyplot.pcolormesh(xsection[::-1,:])
+                pyplot.title('flux: '+str(xsection.sum())+' Sv, or:'+str(totaldrk))
+                pyplot.colorbar()
+                
+                ax4 = pyplot.subplot(412)
+                pyplot.plot(xsection[::-1,:].sum(0))
+                pyplot.title('total: '+str(xsection.sum())+' Sv')    
+                                    
+                ax2 = pyplot.subplot(413)
+                pyplot.pcolormesh(drakexsectArea.mean(2)[::-1,:]*1.e-6)
+                pyplot.title('Crossectional Area')
+                pyplot.colorbar()
+                
+                ax3 = pyplot.subplot(414)
+                v = np.ma.masked_where(drakexsectArea.mask,velo[t])
+                pyplot.pcolormesh(v.sum(2)[::-1,:])
+                pyplot.title('Velocity')
+                pyplot.colorbar()
+                                                                
+                pyplot.show()
```

Comment:
you sure you don't want to savefig here?

Reply:
![image](https://user-images.githubusercontent.com/4180685/186100423-1b89b405-0011-48e3-9dbd-60a8e80ad6d9.png)